### DISCUSSION TITLE
Using egui to write to a process's stdin?

### DISCUSSION THREAD

**Discussion by melody-rs** [emilk/egui#2845] [Q&A]
I'd like to make an egui textbox that directly writes to a process's stdin, but I'm unsure how to go about doing that?

**Comment by MGlolenstine** [comment]
Do you spawn your process?
If you do, you can use [`Command::spawn`](https://doc.rust-lang.org/std/process/struct.Command.html#method.spawn) to spawn it and capture the `stdin`, `stdout` and `stderr`.
Then you simply unwrap and write to `child.stdin`.

### Example
```rust
let result_child = std::process::Command::new("nc").spawn();
if let Ok(child) = result_child {
    if let Some(mut stdin) = child.stdin{
        stdin.write(b"Hello World!").expect("Failed to write to the process!");
    }
}
```
Example can be seen (but not run) [here](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=d612e84d5f8cfdc544044ab5385a8abc).
